core/Concurrency/LockBackend/MySqlLockBackend.php

```diff
@@ -0,0 +1,165 @@
+<?php
+/**
+ * Piwik - free/libre analytics platform
+ *
+ * @link http://piwik.org
+ * @license http://www.gnu.org/licenses/gpl-3.0.html GPL v3 or later
+ *
+ */
+
+namespace Piwik\Concurrency\LockBackend;
+
+
+use Piwik\Common;
+use Piwik\Concurrency\LockBackend;
+use Piwik\Db;
+use Piwik\DbHelper;
+
+class MySqlLockBackend implements LockBackend
+{
+    const TABLE_NAME = 'locks';
+
+    private static $tableCreated = false;
+
+    public function install()
+    {
+        if (self::$tableCreated) {
+            return;
+        }
+
+        $tableName = self::getTableName();
+
+        DbHelper::createTable($tableName, "
```

Comment:
As the table is installed on demand... If there's a race condition, should we maybe catch a "table already exists" error and fail silently? Or we just install the table the regular way and provide an update for it as well? IT be fine if table is not needed by most users.

Reply:
I thought the SQL was 'CREATE IF NOT EXISTS', but I'll double check.

Reply:
Looks like it's not `CREATE IF NOT EXISTS`, but the exception is already caught: https://github.com/matomo-org/matomo/blob/3.x-dev/core/Db/Schema/Mysql.php#L452